modules/boss.lua

```diff
@@ -30,6 +40,36 @@ local function BossAttempt(event_name, boss_name)
   end
 end
 
+local function EncounterAttempt(event_name, encounter_id)
+  Debug("New Encounter attempt: %s %s", event_name, encounter_id)
+  -- Temporary fix since we cannot unregister DBM callbacks
+  if not mod:IsEnabled() then return end
+
+  local encounter = Encounters:GetEncounter(encounter_id)
```

Comment:
`Encounters:GetEncounter()` is written that it will return one of two things:
1. The encounter data (name and instance_id)
2. `nil`

Doesn't matter what you feed it: `abc`, `nil`, `-1`, `0` these will all result in `return nil`. Only valid encounter_id's will return something else then `nil`.

I agree that the validation in `EncounterAttempt` can be better, will include that in my update.